### ISSUE TITLE
Jet Smart Filters not working on Custom Taxonomy Archive Template

### ISSUE THREAD

**Issue by standiers** [Crocoblock/suggestions#1217]
**Describe the bug**
Posts with custom taxonomies can be filtered on normal pages but when they are on a custom taxonomy archive template, the jet smart filters do not work at all, including the sort filter.
For example my post have business type and business location as custom taxonomy. When I create an archive page for business types, I cannot filter by location within that archive.

**Expected behavior**
For a business listing, I would expect to be able to select a custom category to get an archive page displaying the post related to that category. Then at least be able to apply sorting to navigate the grid of results.

(I am working locally so cannot share URL)

**Comment by standiers** [comment]
It seems to work for anything else than Ajax or mixed method of filtering.

**Comment by Chrimby** [comment]
Have you found a workaround for that? I can confirm: Filter only works for custom post types with "Page reload".

**Comment by standiers** [comment]
The only way for me still is to use the page reload filter action to apply filters. Not great but it works. Hoping for a fix..

**Comment by TspeakDesign** [comment]
I have the same issue. Have just started to try and find a solution but seeing this is not a good start. Will update if I find a way to make this work. I will most likely go with a different pluggin.

**Comment by leandroseja** [comment]
Hello friends, can you share the solution?

**Comment by leandroseja** [comment]
> A única maneira para mim ainda é usar a ação de filtro de recarregamento de página para aplicar filtros. Não é ótimo, mas funciona. Torcendo para uma correção..

Friend, can you share your solution?

**Comment by TspeakDesign** [comment]
> > A única maneira para mim ainda é usar a ação de filtro de recarregamento de página para aplicar filtros. Não é ótimo, mas funciona. Torcendo para uma correção..
> 
> Friend, can you share your solution?

I was unable to resolve the issue myself with the Crocoblock plugin. I have not completed the project, but my plan moving forward will be to use a different plugin option for the filters.

**Comment by leandroseja** [comment]
I'm trying via GET Params and without success.
Can you share your solution?

**Comment by NickSick** [comment]
In my case filters doesn't work in custom post types created with jetengine, even if i disable ajax and mixed mode. 
So confusing and time consuming

**Comment by OutdoorEdLLC** [comment]
I have the same issue - for me the Checkbox Filters work on a JetEngine CPT on an Elementor archive page when I am logged in, but they don't work as an anonymous user.

**Comment by TspeakDesign** [comment]
Yep, I suggest you look for other solutions. This is one limitation of the Croco out of the box.On Feb 20, 2023, at 1:40 AM, Rick Curtis ***@***.***> wrote:﻿
I have the same issue - for me the Checkbox Filters work on a JetEngine CPT on an Elementor archive page when I am logged in, but they don't work as an anonymous user.

**Comment by Mat-Alex** [comment]
> I have the same issue - for me the Checkbox Filters work on a JetEngine CPT on an Elementor archive page when I am logged in, but they don't work as an anonymous user.

Yes I had that issue but I fixed it!

It is nothing to do with crocoblock/jetengine, it is another plugin.

If you inspect the logged out vs logged in pages in the browser console, you will see a bunch of javascript errors all relating to jquery not loading.

If you inspect the source you will see in the logged out version the jquery script has the 'defer' attribute.

Something is at work deferring the script.

In my case it is the (siteground hosting) SG Optimiser plugin>frontend>javascript>'Defer Render-blocking JavaScript' option.

Turned it off.

Strange how it is not applied if logged in??

If this doesn't work for you then maybe it is server side (eg. a cloudflare setting), or a different installed plugin (turn them all off, turn them on again until you find it)

**Comment by Mat-Alex** [comment]
> **Describe the bug** Posts with custom taxonomies can be filtered on normal pages but when they are on a custom taxonomy archive template, the jet smart filters do not work at all, including the sort filter. For example my post have business type and business location as custom taxonomy. When I create an archive page for business types, I cannot filter by location within that archive.
> 
> **Expected behavior** For a business listing, I would expect to be able to select a custom category to get an archive page displaying the post related to that category. Then at least be able to apply sorting to navigate the grid of results.
> 
> (I am working locally so cannot share URL)

If this is related to *sorting* on a taxonomy, you can't do that with WP.

You have to store the data as custom fields instead, although that does take a hit on performance. Doesn't include repeater fields though: JetSmart filters don't recognise them: FEATURE REQUEST!!

https://metabox.io/custom-fields-vs-custom-taxonomies/

**Comment by NefiCaldeira** [comment]
**I solved it as follows, I defined the ID of the Loop grid Element (With my Custom Layout) as result (Note: you cannot define the section/container depending on the version of Elementor) so it worked, my theory for what happens is that defining the Container as where the result will be received, JetEngine tries to process it and ends up getting too big to process and cuts the processing and displays the error in question... I used Loop Grid because I was creating an Archive for the products...**

Follow the following steps:

1. Enter JetSmartFilter > settings
2. Go to JetSmartFilters > Indexers Settings
3. Enable Products and save
4. Follow the Elementor screenshots below

![image](https://github.com/Crocoblock/suggestions/assets/103417242/96d61771-0db5-4014-945b-78cc83dca302)
![image](https://github.com/Crocoblock/suggestions/assets/103417242/b6a7596c-245e-4179-8e1e-35bd429e61e3)
**### Elementor Configs**

![image](https://github.com/Crocoblock/suggestions/assets/103417242/f5e24c55-b236-4d14-865f-85b8d25eb4e9)
![image](https://github.com/Crocoblock/suggestions/assets/103417242/3876ab7c-0367-4338-bac7-24fdec159160)
